consensus/polybft/extra.go

```diff
@@ -142,14 +142,77 @@ func (i *Extra) UnmarshalRLPWith(v *fastrlp.Value) error {
 	return nil
 }
 
-// ValidateBasic contains extra data basic set of validations
-func (i *Extra) ValidateBasic(parentExtra *Extra) error {
+// ValidateFinalizedHeader contains extra data validations for finalized headers
+func (i *Extra) ValidateFinalizedHeader(header *types.Header, parent *types.Header, parents []*types.Header,
+	chainID uint64, consensusBackend polybftBackend, logger hclog.Logger) error {
+	// validate committed signatures
+	blockNumber := header.Number
+	if i.Committed == nil {
+		return fmt.Errorf("failed to verify signatures for block %d because signatures are not present", blockNumber)
+	}
+
+	checkpointHash, err := i.Checkpoint.Hash(chainID, header.Number, header.Hash)
```

Comment:
Since you are checking if checkpoint is `nil` in some functions, I would say, check it here as well.

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1208/commits/e1b8a65f095fb8167da7c8befe45153449b3b997